devops/poseidon-deployer/contracts/overkill.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {PoseidonT2, PoseidonT3, PoseidonT4, PoseidonT5, PoseidonT6, PoseidonT7} from "./poseidon.sol";
+
+import {Multicallable} from "./Multicallable.sol";
+import {LibClone} from "./LibClone.sol";
+
+// deploy all the Poseidon contracts in one tx
+// try to break that down into tasks
+
+contract PoseidonFactory is Multicallable {
+    /// -----------------------------------------------------------------------
+    /// Library Usage
+    /// -----------------------------------------------------------------------
+
+    using LibClone for address; //hacks
+
+    /// -----------------------------------------------------------------------
+    /// Events
+    /// -----------------------------------------------------------------------
+    
```

Comment:
🧐 think we missed T1

Reply:
T1 makes number of input to 0 (i.e. no input) due to [`const t = nInputs + 1;`](https://github.com/iden3/circomlibjs/blob/main/src/poseidon_gencontract.js#L26)